packages/flutter_test/lib/src/window.dart

```diff
@@ -944,3 +982,263 @@ class TestPlatformDispatcher implements ui.PlatformDispatcher {
+  // ignore: avoid_setters_without_getters
+  set _viewConfigurationLegacy(ui.ViewConfiguration? viewConfiguration) {
+    if (viewConfiguration == null) {
+      _overrideConfiguration = null;
+    } else {
+      _overrideConfiguration = viewConfiguration.copyWith(view: this);
+    }
+    platformDispatcher.onMetricsChanged?.call();
+  }
+
+  @override
+  Object get viewId => _view.viewId;
+
+  @override
+  /// The device pixel ratio to use for this test.
+  ///
+  /// Defaults to the value provided by [ui.FlutterView.devicePixelRatio]. This
+  /// can only be set in a test environment to emulate different view
+  /// configurations. A standard [ui.FlutterView] is not mutable from the framework.
+  ///
+  /// See also:
+  ///
+  ///   * [ui.FlutterView.devicePixelRatio] for the standard implementation
+  ///   * [resetDevicePixelRatio] to reset this value specifically
+  ///   * [reset] to reset all test values for this view
+  double get devicePixelRatio => _viewConfiguration.devicePixelRatio;
+  set devicePixelRatio(double devicePixelRatio) {
+    _viewConfiguration = _viewConfiguration.copyWith(devicePixelRatio: devicePixelRatio);
+    platformDispatcher.onMetricsChanged?.call();
+  }
+
+  /// Resets [devicePixelRatio] for this test view to the default value for this view.
+  void resetDevicePixelRatio() => devicePixelRatio = _view.devicePixelRatio;
+
+  @override
+  /// The display features to use for this test.
+  ///
+  /// Defaults to the value provided by [ui.FlutterView.displayFeatures]. This
+  /// can only be set in a test environment to emulate different view
+  /// configurations. A standard [ui.FlutterView] is not mutable from the framework.
+  ///
+  /// See also:
+  ///
+  ///   * [ui.FlutterView.displayFeatures] for the standard implementation
+  ///   * [resetDisplayFeatures] to reset this value specifically
+  ///   * [reset] to reset all test values for this view
+  List<ui.DisplayFeature> get displayFeatures => _viewConfiguration.displayFeatures;
+  set displayFeatures(List<ui.DisplayFeature> displayFeatures) {
+    _viewConfiguration = _viewConfiguration.copyWith(displayFeatures: displayFeatures);
+    platformDispatcher.onMetricsChanged?.call();
+  }
+
+  /// Resets [displayFeatures] to the default values for this view.
+  void resetDisplayFeatures() => displayFeatures = _view.displayFeatures;
+
+  @override
+  /// The padding to use for this test.
+  ///
+  /// Defaults to the value provided by [ui.FlutterView.padding]. This
+  /// can only be set in a test environment to emulate different view
+  /// configurations. A standard [ui.FlutterView] is not mutable from the framework.
+  ///
+  /// See also:
+  ///
+  ///   * [ui.FlutterView.padding] for the standard implementation
+  ///   * [resetPadding] to reset this value specifically
+  ///   * [reset] to reset all test values for this view
+  ui.ViewPadding get padding => _viewConfiguration.padding;
+  set padding(ui.ViewPadding padding) {
```

Comment:
All the docs with setters that take ViewPadding should probably link to `TestViewPadding` as a thing to use. Maybe we could even change the Type here to only take TestViewPadding to guide people? Not sure if dart allows that.

Reply:
The  "only take TestViewPadding" suggestion didn't work when I tried it. That's what I wanted to do as well... Good call on adding a reference to `TestViewPadding` in the docs, I'll add that in.

Reply:
I decided to force the "only take FakeViewPadding" path to work. It means I have to wrap the default value (because the getter has to match the setter), but I think it'll make the use of the test classes much clearer, so worth it in my opinion.